### PR TITLE
Feature/add http header generator

### PR DESCRIPTION
**Pull request by 1Jo1**
-Add Http Header Generator to avoid being block
-You have to override the property User-Agent explicitly because of the lib Colly

### CODE REVIEW

File: http-header-generator/generator.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	"math/rand"
+	"net/http"
+)
+
+type HttpHeaderGenerator struct {
+	browserAgent BrowserAgent
+}
+
+func New() *HttpHeaderGenerator {
+	httpHeader := &HttpHeaderGenerator{}
```
Comment (alexmorten) [inline http-header-generator/generator.go]: `generator` would be easier to understand here

File: scraper/scraper.go [hunk 2]
```diff
@@ -92,11 +95,11 @@ func (s *Scraper) Close() {
 }
 
 //ScrapeUserFollowGraph returns the follow information for a userName
```
Comment (alexmorten) [inline scraper/scraper.go]:
```suggestion
//scrapeUserFollowGraph returns the follow information for a userName
```

**Review by alexmorten — CHANGES_REQUESTED** [review]
Looking good!

Please add the `useragents.json` to the Dockerfile of the scraper. Otherwise, just two tiny things

File: scraper/Dockerfile [hunk 3]
```diff
@@ -10,5 +10,6 @@ FROM alpine
 RUN apk --no-cache add ca-certificates
 RUN mkdir app
 COPY --from=builder /go/src/github.com/codeuniversity/smag-mvp/instascraper_scraper /app
+COPY --from=builder /go/src/github.com/codeuniversity/smag-mvp/useragents.json /app
```
Comment (alexmorten) [inline scraper/Dockerfile:13]: You don't need the `--from=builder` 
I think the caching is simpler if you just use the useragents.json from the build-context

Comment (1Jo1) [reply scraper/Dockerfile:13]: I will create another PR for that

**Pull request merged**